blocks/library/shortcode/test/index.js

```diff
@@ -2,12 +2,13 @@
  * Internal dependencies
  */
 import { name, settings } from '../';
+import { Shortcode } from '../block';
 import { blockEditRender } from 'blocks/test/helpers';
 
+settings.edit = Shortcode;
```

Comment:
This change seems weird to me but I'm not familiar with these tests enough to understand whether it's legit or not? cc @gziolo  Can you clarify the intention?

Reply:
I think it is to avoid having `edit` wrapped with `connect`:

```js
const applyConnect = connect(
	( state ) => {
		return {
			postId: getCurrentPostId( state ),
		};
	},
);```

I think it can be coded differently to avoid this kind of workarounds. It seems like most of the blocks shouldn't depend on the external data when they load for the first time. Shortcode isn't an exception here.

Reply:
Right, I missed this, we should definitely avoid using the selector directly here since it's on another module, maybe we could use the reusable block temporary trick for now `state.currentPost.id` to avoid messing with the bundles. 

Hopefully, we'd agree at some point at merging Editor and Blocks modules to fix all these issues.

Reply:
As rightly mentioned by @gziolo, this change was done because the jest tests avoid having a connected component. I've described the issue in detail in this <a href="https://github.com/WordPress/gutenberg/pull/4710#issuecomment-361572324">comment </a>.